graylog2-server/src/main/java/org/graylog2/migrations/V202201281638_GeoIpResolverConfigMigration.java

```diff
@@ -17,24 +17,39 @@
 
 package org.graylog2.migrations;
 
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.auto.value.AutoValue;
 import com.mongodb.client.MongoCollection;
 import com.mongodb.client.model.Filters;
 import com.mongodb.client.model.Updates;
 import com.mongodb.client.result.UpdateResult;
 import org.bson.Document;
 import org.bson.conversions.Bson;
+import org.graylog.autovalue.WithBeanGetter;
 import org.graylog.plugins.map.config.DatabaseVendorType;
 import org.graylog.plugins.map.config.GeoIpResolverConfig;
 import org.graylog2.database.MongoConnection;
+import org.graylog2.plugin.cluster.ClusterConfigService;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.inject.Inject;
+import java.time.ZoneId;
 import java.time.ZonedDateTime;
 
-public class V20211221144300_GeoIpResolverConfigMigration extends Migration {
+/**
+ * The Original migration had a bug which caused the <b>db_vendor_type</b>  field to always be updated--replacing any user supplied values.
+ *
+ * <p>
+ * This update writes a completion status record upon initial successful migration, and checks for presence of such record
+ * before stating any subsequent migration--aborts if record found.
+ * </p>
+ */
+public class V202201281638_GeoIpResolverConfigMigration extends Migration {
 
-    private static final Logger LOG = LoggerFactory.getLogger(V20211221144300_GeoIpResolverConfigMigration.class);
+    private static final Logger LOG = LoggerFactory.getLogger(V202201281638_GeoIpResolverConfigMigration.class);
```

Comment:
My original suggestion to rename the migration was to cause it to run again for any devs who have run it already, but that was invalid, because the migration was not previously writing can checking the `MigrationCompleted` object. Can the rename be reverted, since it's not needed? That might make the PR changeset more concise if it needs to be reviewed in the future.

Reply:
Reverted name.